core/services/workflows/syncer/handler.go

```diff
@@ -359,8 +362,9 @@ func (h *eventHandler) workflowRegisteredEvent(
 	}
 
 	// Ensure that there is no running workflow engine for the given workflow ID.
-	if h.engineRegistry.Contains(hex.EncodeToString(payload.WorkflowID[:])) {
-		return fmt.Errorf("workflow is already running, so not starting it : %s", hex.EncodeToString(payload.WorkflowID[:]))
+	engineKey := h.engineRegistry.KeyFor(payload.WorkflowOwner, payload.WorkflowName)
+	if h.engineRegistry.Contains(engineKey) {
+		return fmt.Errorf("workflow is already running, so not starting it : %s", wfID)
```

Comment:
This is incorrect -- an engine could be running but with a different workflow ID

Reply:
@cedric-cordenier @justinkaseman can you guys explain the need to not use workflow ID as the key for an engine?